backend/src/services/html_pdf_renderer.py

```
@@ -0,0 +1,53 @@
+        )
+
+    def render_html(self, template_name: str, context: Dict[str, Any]) -> str:
+        template = self.env.get_template(template_name)
+        return template.render(**context)
+
+    def render_pdf(self, html_content: str, output_path: Path) -> Path:
+        """Render given HTML content to PDF using Playwright. Raises if Playwright not available."""
+        try:
+            from playwright.sync_api import sync_playwright
+        except Exception as e:
+            raise RuntimeError("Playwright is not installed. Run: python -m playwright install chromium") from e
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)